### PR TITLE
RSDK-9037: Add AttachDirectionalAwareness to DoCommand

### PR DESCRIPTION
**Pull request by martha-johnston**
I'm not sure I think this is any better than the way it was before. I considered the method Olivia suggested by converting the `DoCommand` interface into a motor object, but it couldn't be `motor.Motor`, it would have to be specifically `gpio.Motor` because the interface doesn't implement all the API functions. And imo `gpio.Motor` in `DoCommand` is no better than `single.Encoder` in `setup.go` as it was before. Open to ideas tho, because I don't love this implementation. 

Basically, my goal here was to avoid casting to any specific model types of either motor or encoder. I did have to cast to the `DirectionAware` type, but that seemed better since it's specific to single encoder.

### CODE REVIEW

File: components/motor/gpio/encoded.go [hunk 1]
```diff
@@ -88,8 +88,8 @@ func newEncodedMotor(
 	}
 
 	if em.rampRate == 0 || em.rampRate > 0.5 {
-		em.rampRate = 0.05 // Use a conservative value by default.
 		em.logger.Warnf("setting ramp rate to default value of 0.05 instead of %v", em.rampRate)
+		em.rampRate = 0.05 // Use a conservative value by default.
```
Comment (martha-johnston) [inline components/motor/gpio/encoded.go:92]: this I just happened to see when I was testing and before it would always print out "setting ramp rate to default value of 0.05 instead of 0.05" so I changed it

File: components/motor/gpio/setup.go [hunk 2]
```diff
@@ -213,7 +220,7 @@ func createNewMotor(
 				return nil, err
 			}
 		default:
-			m, err = setupMotorWithControls(ctx, basic, e, cfg, logger)
+			m, err = setupMotorWithControls(ctx, m, e, cfg, logger)
```
Comment (martha-johnston) [inline components/motor/gpio/setup.go:223]: all the changes in controlled motor come from this change. I wanted to remove the basic casting and try to keep everything at the motor interface level

**Review by JohnN193 — APPROVED** [review]
LGTM, we spoke offline and it seemed like this was worth doing because it might let motors defined in other languages use single encoder
